indico/modules/events/contributions/util.py

```diff
@@ -72,3 +82,125 @@ def serialize_contribution_person_link(person_link):
+
+    def build_query(self):
+        timetable_entry_strategy = joinedload('timetable_entry')
+        timetable_entry_strategy.lazyload('*')
+        return (self.report_event.contributions
+                .filter_by(is_deleted=False)
+                .order_by(Contribution.friendly_id)
+                .options(timetable_entry_strategy))
+
+    def filter_report_entries(self, query, filters):
+        if not filters.get('items'):
+            return query
+        item_criteria = []
+        if 'session' in filters['items']:
+            session_ids = filters['items']['session']
+            item_criteria.append(Contribution.session_id.in_(session_ids))
+        if 'track' in filters['items']:
+            track_ids = filters['items']['track']
+            item_criteria.append(Contribution.track_id.in_(track_ids))
+        if 'type' in filters['items']:
+            type_ids = filters['items']['type']
+            item_criteria.append(Contribution.type_id.in_(type_ids))
+        # TODO: Handle contribution status
+        return query.filter(db.or_(*item_criteria))
+
+    def get_contrib_report_args(self):
+        contributions_query = self.build_query()
+        total_entries = contributions_query.count()
+        contributions = self.filter_report_entries(contributions_query, self.report_config['filters']).all()
+        sessions = [{'id': s.id, 'title': s.title} for s in self.report_event.sessions.filter_by(is_deleted=False)]
+        tracks = [{'id': int(t.id), 'title': to_unicode(t.getTitle())}
+                  for t in self.report_event.as_legacy.getTrackList()]
+        return {'contribs': contributions, 'sessions': sessions, 'tracks': tracks, 'total_entries': total_entries}
+
+    def render_contrib_report(self, contrib=None):
+        """Render the contribution report template components.
+
+        :param contrib: Used in RHs responsible for CRUD operations on a
+        contribution.
+        :returns: dict - dict containing the report's entries, the fragment of
+        displayed entries and whether the contrib passed is displayed in the
+        results.
+        """
+        contr_report_args = self.get_contrib_report_args()
+        total_entries = contr_report_args.pop('total_entries')
+        tpl = get_template_module('events/contributions/management/_contribution_report.html')
+        fragment = tpl.render_displayed_entries_fragment(len(contr_report_args['contribs']), total_entries)
+        return {'html': tpl.render_contrib_report(event=self.report_event, **contr_report_args),
+                'displayed_records_fragment': fragment,
+                'hide_contrib': contrib not in contr_report_args['contribs'] if contrib else None}
+
+    def flash_info_message(self, contrib, hide_contrib):
+        if hide_contrib:
+            flash(_("The contribution '{}' is not displayed in the list due to the enabled filters")
+                  .format(contrib.title), 'info')
+
+    def get_report_url(self, uuid=None):
```

Comment:
not sure if it'ss much clearer than the separate `url_for` calls but i'm fine with either way